src/ucp/api/ucp.h

```diff
@@ -3135,6 +3108,189 @@ ucs_status_ptr_t ucp_worker_flush_nb(ucp_worker_h worker, unsigned flags,
+   *                       data will persist after the callback returns.
+   * @param [in]  remaining_length Length of data which has not yet been
+   *                       transferred
+   * @param [out] recv     Struture passed to the ucp library to control
+   *                       placement of the remaining data
+   *
+   * @return UCS_OK        @a data will not persist after the callback returns.
+   *
+   * @return UCS_INPROGRESS Can only be returned if flags is set to
+   *                        UCP_CB_PARAM_FLAG_DATA. If UCP_INPROGRESS
+   *                        is returned, data will persist after the
+   *                        callback has returned. To free the memory,
+   *                        a pointer to the data must be passed into
+   *                        @ref ucp_am_data_release.
+   *
+   * @return UCS_ERR_NO_MEMORY Callback was unable to allocate memory for
+   *                        Active Message to be received into
+   *
+   *
+   * @note This callback should be set and released
+   *       by @ref ucp_worker_set_am_handler function.
+   *
+   */
+typedef ucs_status_t (*ucp_am_callback_t)(
+                                         void *arg,
+                                         void *data,
+                                         size_t length,
+                                         ucp_ep_h reply_ep,
+                                         unsigned flags,
+                                         size_t remaining_length,
+                                         ucp_am_recv_t *recv
+                                        );
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief Add user defined callback for Active Message.
+ *
+ * This routine installs a user defined callback to handle incoming Active
+ * Messages with a specific id. This callback is called whenever an Active
+ * Message that was sent from the remote peer by @ref ucp_am_send_nb
+ * is received on this worker
+ *
+ * @param [in]  worker      UCP worker on which to set the Active Message
+ *                          handler.
+ * @param [in]  params      Tuning parameters for the active message
+ * @param [in]  id          Active Message id.
+ * @param [in]  cb          Active Message callback. NULL to clear.
+ * @param [in]  arg         Active Message argument, which will be passed
+ *                          in to every invocation of the callback as the
+ *                          arg argument.
+ *
+ * @return error code if the worker does not support Active Messages or
+ *         requested callback flags.
+ */
+ucs_status_t ucp_worker_set_am_handler(
+                                       ucp_worker_h worker,
+                                       const ucp_am_params_t *params,
+                                       uint16_t id,
+                                       ucp_am_callback_t cb,
+                                       void *arg
+                                       );
```

Comment:
- this breaks backward compatibility 
- lines are not wrapped according to code style, pls align the style with other functions in UCP